core/src/main/scala/kafka/server/metadata/DynamicClientQuotaPublisher.scala

```diff
@@ -48,6 +51,11 @@ class DynamicClientQuotaPublisher(
   ): Unit = {
     val deltaName = s"MetadataDelta up to ${newImage.highestOffsetAndEpoch().offset}"
     try {
+        val clientQuotaCallback = conf.getConfiguredInstance(QuotaConfig.CLIENT_QUOTA_CALLBACK_CLASS_CONFIG, classOf[ClientQuotaCallback])
```

Comment:
we should use the callback in `quotaManagers` rather than creating a new one! They are different instances and so this approach can update the callback used by `quotaManagers`